### DISCUSSION TITLE
How to enable WebGL support?

### DISCUSSION THREAD

**Discussion by Ravenwater** [McShelby/hugo-theme-relearn#457] [Q&A] [answered]
Trying to add WebGL animations to Hugo pages and not having any luck. What is the mechanism I should leverage to add a webgl portal and animation program to a page? 

I tried a partial by adding a layouts/partial/my-viz.html with the WebGL animation, but that not getting picked up. Also seeing `type` frontmatter that could apply, but I do not see any examples that demonstrate the specifics.

What is the proper architecture for adding WebGL animations to a Relearn based site?

**Comment by McShelby** [comment]
I am not quite sure what result you want to accomplish. If you are using this theme there would always be the menu and nav bar around. The best way here would probably be a shortcode.

Otherwise it's maybe best to ask your question in the [official Hugo forum](https://discourse.gohugo.io/) as I have the feeling this is not related solely to this theme.

**Comment by Ravenwater** [comment, marked as answer]
I agree, this is not specific to Relearn, I just had hoped that somebody had already figured this out for this theme.

Normally, a WebGL animation is tied to a drawing canvas, but even when I include the webgl code right into the markdown it seems to completely ignore it, so I am missing something foundational.

I had this working 6 years ago, but something has changed in the Hugo stack that is tripping me up. It is definitely something that needs to be properly architected. I was using partials last time but found one solution and reused it leading to some scalability issues. I would like to fix this this time around. 

So weird that I would be the only one that wants to put an interactive 3D visualization in a Hugo documentation site. When I google the topic, the only hit I get is my own request from 2017.

**Reply by McShelby** [reply]
It's an interesting use case. Although I guess the combination of technologies used here are making it pretty unique (Hugo + WebGL).

With latest Hugo versions to include HTML inside your markdown you need at least `[markup.goldmark.renderer] unsafe=true` in your `config.toml` (see [Hugo docs](https://gohugo.io/getting-started/configuration-markup/#goldmark) for the exact syntax). Maybe this helps a bit.

**Comment by Ravenwater** [comment]
Asked: https://discourse.gohugo.io/t/looking-for-the-right-architecture-to-enable-interactive-webgl-animations-in-a-hugo-site/42716

**Comment by Ravenwater** [comment]
@McShelby that was an excellent tidbit of info. It indeed seems to have just been ripping the code right out. With your pointer, I now at least are getting console errors indicating missing libs, but that means now something is happening... let's see how far I get.

**Reply by McShelby** [reply]
Don't hesitate to ask. Also the guys in the official forum are really kind and helpful.

**Comment by Ravenwater** [comment]
Tada!!!

![image](https://user-images.githubusercontent.com/466641/215621332-bb3b952e-3cc6-483c-bb20-aacf0b4b9caf.png)

**Comment by McShelby** [comment]
Also, for an easier transition from older versions of Relearn or even Learn, see the [migration page](https://mcshelby.github.io/hugo-theme-relearn/basics/migration/index.html)
